src/components/Clause/index.js

```diff
@@ -164,19 +164,21 @@ const ClauseComponent = (props) => {
             </S.DeleteWrapper>
           </>
         }
-        <S.ClauseBody {...props.attributes}>
+        <S.ClauseBody {...props.attributes} ref={ref}>
```

Comment:
Curious why the `ref` was needed here, also why here instead of `ClauseContext.Provider` or `S.ClauseWrapper`

Reply:
It's needed here because when we call `renderElement` with a functional component, we need to forward the `ref`. When the element is rendered inline like in markdown editor, it's included with `attributes`, so we want it where `props.attributes` gets spread so that it can be used by Slate. This is to fix this warning `Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?`